transcendence-app/src/chat/chatroom/infrastructure/db/postgres/chatroom.postgres.repository.ts

```diff
@@ -60,20 +66,49 @@ export class ChatroomPostgresRepository
 
   async getPaginatedChatrooms(
     paginationDto: Required<PaginationWithSearchQueryDto>,
-  ): Promise<Chatroom[] | null> {
-    const { limit, offset, sort, search } = paginationDto;
-    const orderBy =
-      sort === BooleanString.True ? ChatroomKeys.NAME : ChatroomKeys.ID;
-    const chatroomsData = await makeQuery<Chatroom>(this.pool, {
-      text: `SELECT *
-             FROM ${this.table}
-             WHERE ${ChatroomKeys.NAME} ILIKE $1
-             ORDER BY ${orderBy}
+  ): Promise<GenericChat[] | null> {
+    const { limit, offset, search } = paginationDto;
+    const chatroomsData = await makeQuery<GenericChatData>(this.pool, {
+      text: `WITH crMsg AS (SELECT m."chatroomId",
+                                   m."userId",
+                                   m."id",
+                                   m."createdAt"
+                            FROM ${table.CHATROOM_MESSAGE} m),
+                  crDateProvider AS (SELECT m."chatroomId",
+                                            ROW_NUMBER() OVER (
+                                                PARTITION BY m."chatroomId" ORDER BY m."createdAt" DESC
```

Comment:
maybe I can use the enums of the different "userKeys.ID" and such, so that the IDEs can find the relevant columns, and improve on the code duplication, but this is how far I've gotten for now

Reply:
I cannot deduplicate code without giving up on my IDE query parsing. I think it's not worth it, so I'm leaving it just like that. Besides that, I've made the changes in order to use the enums for the columns of the different tables.